tests/extractors/test_bigquery_extractor.py

```diff
@@ -144,16 +135,8 @@ def test_extract_error(self, mock_client, mock_hook):
 
         bq_extractor = BigQueryExtractor(task)
 
-        steps_meta = bq_extractor.extract()
-
-        assert steps_meta[0].name == "TestBigQueryExtractorE2E.task_id"
-        assert steps_meta[0].inputs == []
-        assert steps_meta[0].outputs == []
-        assert steps_meta[0].context is not None
-        assert steps_meta[0].context["sql"] == task.sql
-        assert steps_meta[0].context["bigquery.sql.parsed.inputs"] \
-            == '["customers"]'
-        assert steps_meta[0].context["bigquery.sql.parsed.outputs"] == '[]'
+        steps_meta_extract = bq_extractor.extract()
```

Comment:
I would expect this test to call `extract_on_complete` since it is testing the logic when the BQ client throws an exception. (was there a mistake before?)
Maybe just modify it to actually test the error case when the client throws an exception in extract_on_complete?

Reply:
The test looks right, it's just validating that the first extract doesn't generate the error but extract_on_complete does. extract_on_complete gets called a few lines after this.